flutter_news_example/lib/app/view/app.dart

```diff
@@ -65,17 +64,15 @@ class AppView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      themeMode: ThemeMode.light,
-      theme: const AppTheme().themeData,
-      darkTheme: const AppDarkTheme().themeData,
-      localizationsDelegates: AppLocalizations.localizationsDelegates,
-      supportedLocales: AppLocalizations.supportedLocales,
-      home: AuthenticatedUserListener(
-        child: FlowBuilder<AppStatus>(
-          state: context.select((AppBloc bloc) => bloc.state.status),
-          onGeneratePages: onGenerateAppViewPages,
-        ),
+    return RedirectStreamScope(
+      appBloc: context.read<AppBloc>(),
+      child: MaterialApp.router(
+        themeMode: ThemeMode.light,
+        theme: const AppTheme().themeData,
+        darkTheme: const AppDarkTheme().themeData,
+        localizationsDelegates: AppLocalizations.localizationsDelegates,
+        supportedLocales: AppLocalizations.supportedLocales,
+        routerConfig: AppRouter.router(),
```

Comment:
can we create a GoRouter object like a repository, and inject it to be able to test?

Reply:
```
child: MaterialApp.router(
        builder: (context, child) => FeedbackHelper.build(
          context: context,
          child: StreamChat(
            client: getIt<StreamChatClient>(),
            child: StreamChatCustomizedThemeData(
              child: FToastBuilder()(context, child),
            ),
          ),
        ),
        title: 'Queerie',
        theme: getIt<QTTheme>().darkTheme,
        localizationsDelegates: const [
          AppLocalizations.delegate,
          GlobalMaterialLocalizations.delegate,
          GlobalCupertinoLocalizations.delegate,
        ],
        supportedLocales: AppLocalizations.supportedLocales,
        routerConfig: getIt<GoRouter>(),
      ),
```